IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m

```diff
@@ -433,7 +433,36 @@ - (void)decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
         return;
     }
     
-    decisionHandler(WKNavigationActionPolicyAllow);
+    if (self.customHeaderProvider)
+    {
+        [self.customHeaderProvider getCustomHeaders:navigationAction.request
+                                                    forHost:requestURL.host
+                                            completionBlock:^(NSDictionary<NSString *, NSString *> *extraHeaders, __unused NSError *error){
+            if (extraHeaders && extraHeaders.count > 0)
+            {
+                NSMutableURLRequest *newUrlRequest = [navigationAction.request mutableCopy];
+                
+                for (NSString *headerKey in extraHeaders)
+                {
+                    if (![NSString msidIsStringNilOrBlank:extraHeaders[headerKey]])
+                    {
+                        [newUrlRequest setValue:extraHeaders[headerKey] forHTTPHeaderField:headerKey];
```

Comment:
Should we check if the header already exists in the request before adding it?

Reply:
IMO replacing still with new header would help in case of nonce-expired

Reply:
The redirect request has been passed into header provider to decide if they want to overwrite existing header. So once we get back a dictionary, we assume new value will be used to overwrite old values :)